lib/src/ast.dart

```diff
@@ -151,6 +121,15 @@ bool isHashCode(ClassMember element) =>
     (element is FieldDeclaration &&
         getFieldIdentifier(element, 'hashCode') != null);
 
+/// Return true if this compilation unit [node] is declared within the given
+/// [package]'s `lib/` directory.
```

Comment:
To differentiate between, say, "public interface" files immediately in lib/, and source files in, e.g. lib/src/, maybe say "declared within the given package's `lib/` directory _tree_."

Reply:
Good idea.